packages/react-components/src/components/chart/chartOptions/seriesAndYAxis/convertSeriesAndYAxis.ts

```diff
@@ -190,21 +191,37 @@ export const useSeriesAndYAxis = (
     performanceMode,
   }: { styleSettings: StyleSettingsMap; thresholds: Threshold[]; axis?: ChartAxisOptions; performanceMode?: boolean }
 ) => {
-  const defaultYAxis: YAXisComponentOption[] = [convertChartYAxis(axis)];
+  const axisString = JSON.stringify(axis);
+  const defaultYAxis: YAXisComponentOption[] = useMemo(
+    () => [convertChartYAxis(axis)],
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [axisString]
+  );
   const convertedThresholds = convertThresholds(thresholds);
-  const getStyles = getChartStyleSettingsFromMap(styleSettings);
 
-  const { series, yAxis } = datastreams
-    .map((datastream) => convertSeriesAndYAxis(getStyles(datastream), { performanceMode })(datastream))
-    .reduce(reduceSeriesAndYAxis, { series: [], yAxis: defaultYAxis });
+  // Dependency string
+  const datastreamString = JSON.stringify(datastreams);
+  const getStylesString = JSON.stringify(styleSettings);
+
+  const { series, yAxis } = useMemo(() => {
+    return datastreams
+      .map((datastream) =>
+        convertSeriesAndYAxis(getChartStyleSettingsFromMap(styleSettings)(datastream), { performanceMode })(datastream)
+      )
+      .reduce(reduceSeriesAndYAxis, { series: [], yAxis: defaultYAxis });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [datastreamString, defaultYAxis, getStylesString, performanceMode]);
 
   if (series.length > 0) {
     series[0].markArea = convertedThresholds.markArea;
     series[0].markLine = convertedThresholds.markLine;
   }
 
-  return {
-    series,
-    yAxis,
-  };
+  return useMemo(
```

Comment:
arent series and yaxis already memoized above ^ ?

Reply:
updated